libraries/chain/controller.cpp

```diff
@@ -4707,17 +4709,27 @@ struct controller_impl {
       return conf.block_validation_mode == validation_mode::LIGHT || conf.trusted_producers.count(producer);
    }
 
-   bool should_terminate(block_num_type head_block_num) const {
+   bool should_terminate(block_num_type head_block_num, block_num_type lib) const {
```

Comment:
parameter name `head_block_num` is misleading. It is not necessarily the head since it is called in `create_block_state_i`.

I think it would be nicer for this to be a `template<class BS>` function, taking either a `block_state` or `block_state_legacy` as parameter.

Reply:
Renamed to `reversible_block_num`.